pkg/mpc-examples/contracts/04-pause-unpause-controller/PauseUnpauseController.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/ILastCreatedPoolFactory.sol";
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/IManagedPool.sol";
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract PauseUnpauseController is Ownable {
+    uint256 private constant _END_SWAP_FEE_PERCENTAGE = 3e15;
+    uint256 private constant _START_SWAP_FEE_PERCENTAGE = 95e16;
+    uint256 private constant _SWAP_FEE_REDUCATION_DURATION = 3 days;
+
+    IVault private _vault;
+    bytes32 private _poolId;
+
+    constructor(IVault vault, address controllerOwner) {
+        // Get poolId from the factory.
+        bytes32 poolId = IManagedPool(ILastCreatedPoolFactory(msg.sender).getLastCreatedPool()).getPoolId();
+
+        // Verify that this is a real Vault and the pool is registered - this call will revert if not.
+        vault.getPool(poolId);
+
+        // store Vault and PoolId.
+        _poolId = poolId;
+        _vault = vault;
+
+        // Transfer ownership from factory to manager.
+        transferOwnership(controllerOwner);
+    }
+
+    /// === public Getters ===
+
+    function getPoolId() public view returns (bytes32) {
+        return _poolId;
+    }
+
+    function getVault() public view returns (IVault) {
+        return _vault;
+    }
+
+    function getSwapEnabled() public view returns (bool) {
+        return _getPool().getSwapEnabled();
+    }
+
+    function isPoolPaused() public view returns (bool) {
+        return !_getPool().getSwapEnabled();
+    }
```

Comment:
Seems redundant to have both of these. Maybe chat with @baileyspraggins to agree on which function you think makes the most sense